### DISCUSSION TITLE
Improve “Download PDF…” UX

### DISCUSSION THREAD

**Discussion by mattstein** [craftcms/commerce#2480] [Ideas]
The order listing’s **Download PDF...** button should be improved to match the more optimal UX of the order detail page.

## Order Detail 👍

The order detail page is ideal with its PDF download option:

1. If no PDF template/format is configured, no button is shown.
    <img width="921" alt="no-pdf" src="https://user-images.githubusercontent.com/2488775/125144610-02425900-e0e4-11eb-87a1-22f932302803.png">
2. If one PDF template/format is configured, a simple **Download PDF** is shown.
     <img width="921" alt="one-pdf" src="https://user-images.githubusercontent.com/2488775/125144634-16865600-e0e4-11eb-99f2-4c42f0baa072.png">
3. If multiple PDF templates/formats are configured, the button’s primary action is to download whichever is configured as the default, with a dropdown menu for otherwise selecting the preferred format.
    <img width="921" alt="multiple-pdfs" src="https://user-images.githubusercontent.com/2488775/125144650-20a85480-e0e4-11eb-93cd-2bf296053a8d.png">

The user is only presented options that make sense given the configuration.

## Order Listing 😐

The order listing page is less optimal. 

![download-pdf](https://user-images.githubusercontent.com/2488775/125144815-b217c680-e0e4-11eb-827f-215da781f666.png)

It _does_ only display its **Download PDF…** button if there‘s at least one PDF template/format configured, but...

1. It always requires opening the download menu and pressing **Download**.
2. It always prompts you to select the PDF template/format even if there’s only one choice.
3. It prompts for download types that only make sense when multiple orders were selected, which adds unnecessary steps:
    - If only one order was selected and only one PDF template/format is configured, we have to click to open this menu, ignore an irrelevant option, and choose a download type.
    - If only one order was selected, we might download a ZIP file containing only one document which just adds a required step to extract it.

## tl;dr

I’d therefore suggest the following changes for the order listing’s Download PDF button:

1. If only one order is selected from the list and only one PDF template/format is configured, omit the `…` from **Download PDF…**, skip the menu, and download the PDF document.
2. If multiple orders are selected from the list and only one PDF template/format is configured, omit the menu’s **PDF** selection.

This would create a more optimal download flow that’s consistent with the beauty of the order detail page’s PDF downloading experience.

**Comment by WHITE-developer** [comment]
Also, can there be a setting to make "Collated PDF" the default? Maybe even to disable the "ZIP file" option?
